brokerapi/brokers/storage/definition.go

```diff
@@ -0,0 +1,101 @@
+	        "plan_updateable": false,
+	        "metadata": {
+	          "displayName": "Google Cloud Storage",
+	          "longDescription": "A Powerful, Simple and Cost Effective Object Storage Service",
+	          "documentationUrl": "https://cloud.google.com/storage/docs/overview",
+	          "supportUrl": "https://cloud.google.com/support/",
+	          "imageUrl": "https://cloud.google.com/_static/images/cloud/products/logos/svg/storage.svg"
+	        },
+	        "tags": ["gcp", "storage"],
+	        "plans": [
+	          {
+	            "id": "e1d11f65-da66-46ad-977c-6d56513baf43",
+	            "service_id": "b9e4332e-b42b-4680-bda5-ea1506797474",
+	            "name": "standard",
+	            "display_name": "Standard",
+	            "description": "Standard storage class",
+	            "service_properties": {"storage_class": "STANDARD"}
+	          },
+	          {
+	            "id": "a42c1182-d1a0-4d40-82c1-28220518b360",
+	            "service_id": "b9e4332e-b42b-4680-bda5-ea1506797474",
+	            "name": "nearline",
+	            "display_name": "Nearline",
+	            "description": "Nearline storage class",
+	            "service_properties": {"storage_class": "NEARLINE"}
+	          },
+	          {
+	            "id": "1a1f4fe6-1904-44d0-838c-4c87a9490a6b",
+	            "service_id": "b9e4332e-b42b-4680-bda5-ea1506797474",
+	            "name": "reduced-availability",
+	            "display_name": "Durable Reduced Availability",
+	            "description": "Durable Reduced Availability storage class",
+	            "service_properties": {"storage_class": "DURABLE_REDUCED_AVAILABILITY"}
+	          }
+	        ]
+	      }`,
+		ProvisionInputVariables: []broker.BrokerVariable{
+			broker.BrokerVariable{
+				FieldName: "name",
+				Type:      broker.JsonTypeString,
+				Details:   "The name of the bucket. There is a single global namespace shared by all buckets so it MUST be unique.",
+				Default:   "a generated value",
+			},
+			broker.BrokerVariable{
+				FieldName: "location",
+				Type:      broker.JsonTypeString,
+				Default:   "US",
+				Details:   `The location of the bucket. Object data for objects in the bucket resides in physical storage within this region. See https://cloud.google.com/storage/docs/bucket-locations`,
+			},
+		},
+		BindInputVariables: accountmanagers.ServiceAccountBindInputVariables(),
+		BindOutputVariables: append(accountmanagers.ServiceAccountBindOutputVariables(),
+			broker.BrokerVariable{
+				FieldName: "bucket_name",
+				Type:      broker.JsonTypeString,
+				Details:   "Name of the bucket this binding is for",
+			},
+		),
+
+		Examples: []broker.ServiceExample{
+			broker.ServiceExample{
+				Name:            "Basic Configuration",
+				Description:     "Create a nearline bucket with a service account that can create/read/delete the objects in it.",
+				PlanId:          "a42c1182-d1a0-4d40-82c1-28220518b360",
+				ProvisionParams: map[string]interface{}{"location": "us"},
```

Comment:
Isn't a bucket name required for provisioning as well?

Reply:
A bucket name will be auto-generated if not provided. I decided to go that route because with buckets having globally unique names I couldn't guarantee that we'd hold the name for this bucket between tests (especially if someone else tried to run the examples).

Reply:
Makes sense.